src/mjolnir/shortcutbuilder.cc

```diff
@@ -471,6 +457,15 @@ uint32_t AddShortcutEdges(GraphReader& reader,
       bool diff_names = directededge->endnode().tileid() == edge_id.tileid() &&
                         !OpposingEdgeInfoMatches(tile, directededge);
 
+      // Recalculate speed limit to include turn durations
+      uint32_t average_speed = edgeinfo.speed_limit();
```

Comment:
shouldnt be messing with the speed limit here. speed limit is what is on the actual speed limit sign. what you want to do here is look at `directededge->speed` (you'll also want to recalculate for truck costing with `directededge->truckspeed`

we have a bunch of conversion constants in `midgard/constants.h` but we seem to be missing a few:
https://github.com/valhalla/valhalla/blob/master/valhalla/midgard/constants.h#L37

we should add something like in that header:

```c++
constexpr double kKMHtoMetersPerSec = 1000./3600.;
constexpr double kMetersPerSectoKMH = 3600./1000.;
```

then we'd have something like this in your new code:

```c++
double new_duration = length  / (directededge->speed()  * kKMHtoMetersPerSec) + total_turn_duration;
uint32_t new_speed = static_cast<uint32_t>(std::round((length / new_duration) * kMetersPerSectoKHM));

//TODO: actually go fix truck turn durations in sif and the other motor vehicle costing models (taxi, bus, etc)
double new_truck_duration = length  / (directededge->truck_speed()  * kKMHtoMetersPerSec) + total_truck_turn_duration;
uint32_t new_truck_speed = static_cast<uint32_t>(std::round((length / new_truck_duration) * kMetersPerSectoKHM));
```

Reply:
Oh, got it. It actually took me some time to understand that the speed I should update is the one in the `newedge` and is copied there implicitly from `directededge. But I was still wondering why there was also speed_limit in the EdgeInfo. Now it makes more sense